### ISSUE TITLE
Jupyterhub support prespawning of nodes + auto download the needed images

### ISSUE THREAD

**Issue by filippomc** [MetaCell/cloud-harness#374]

**Comment by filippomc** [comment]
https://z2jh.jupyter.org/en/latest/administrator/optimization.html#pulling-images-before-users-arrive this should do the job: just need to configure dynamically to pull the right image(s)

**Comment by zsinnema** [comment]
@filippomc this won't help you because if a new node is spawned it always needs to download the right images to startup the pod the user requested. Thus the user will end up waiting for the download likewise when you do not autopull... 

this only helps in case we spawn a new node way before a new user could arrive.

**Comment by filippomc** [comment]
We can setup the prepuller to pull any image we want with `prePuller.extraImages`

**Comment by filippomc** [comment]
Also jupyterhub has a strategy to scale up before actual users come https://z2jh.jupyter.org/en/latest/administrator/optimization.html#scaling-up-in-time-user-placeholders

**Comment by zsinnema** [comment]
yes if we use the scale up strategy the pre puller will be beneficial for us... in case we don't pre scale up before some one arrives then the pre puller has no use
